hathor/manager.py

```diff
@@ -385,88 +384,91 @@ def _initialize_components(self) -> None:
-                    # TODO: deal with invalid tx
-                    if tx.can_validate_full():
-                        self.tx_storage.add_to_indexes(tx)
-                        assert tx.validate_full(skip_block_weight_verification=skip_block_weight_verification)
-                        self.consensus_algorithm.update(tx)
-                        self.tx_storage.indexes.mempool_tips.update(tx)
-                        self.step_validations([tx])
-                    else:
-                        assert tx.validate_basic(skip_block_weight_verification=skip_block_weight_verification)
-                    self.tx_storage.save_transaction(tx, only_metadata=True)
-                else:
-                    # TODO: deal with invalid tx
-                    if not tx_meta.validation.is_final():
-                        if not tx_meta.validation.is_checkpoint():
-                            assert tx_meta.validation.is_at_least_basic(), f'invalid: {tx.hash_hex}'
-                    elif tx.is_transaction and tx_meta.first_block is None and not tx_meta.voided_by:
-                        assert self.tx_storage.indexes is not None
-                        self.tx_storage.indexes.mempool_tips.update(tx)
-                    self.tx_storage.add_to_indexes(tx)
-                    if tx.is_transaction and tx_meta.voided_by:
-                        self.tx_storage.del_from_indexes(tx)
-            except (InvalidNewTransaction, TxValidationError):
-                self.log.error('unexpected error when initializing', tx=tx, exc_info=True)
-                raise
-
-            if tx.is_block:
-                block_count += 1
-
-                # this works because blocks on the best chain are iterated from lower to higher height
-                assert tx.hash is not None
-                assert tx_meta.validation.is_at_least_basic()
-                if not tx_meta.voided_by and tx_meta.validation.is_fully_connected():
-                    # XXX: this might not be needed when making a full init because the consensus should already have
-                    self.tx_storage.indexes.height.add_reorg(tx_meta.height, tx.hash, tx.timestamp)
-
-                # Check if it's a checkpoint block
-                if tx_meta.height in checkpoint_heights:
-                    if tx.hash == checkpoint_heights[tx_meta.height]:
-                        del checkpoint_heights[tx_meta.height]
-                    else:
-                        # If the hash is different from checkpoint hash, we stop the node
-                        self.log.error('Error initializing the node. Checkpoint validation error.')
-                        sys.exit()
-            else:
-                tx_count += 1
-
-            if time.time() - t2 > 1:
-                dt = LogDuration(time.time() - t2)
-                self.log.warn('tx took too long to load', tx=tx.hash_hex, dt=dt)
+        # FIXME: full-verification does not work alongside the new index initialization
```

Comment:
Has it been fixed? There's a lot of commented code after this comment.

Reply:
I guess it was fixed, right? Just double checking...

Reply:
Yes, full-verification now works again.